File: templates/ios/Result.m
#import "{{config.classPrefix}}{{name}}.h"

{{#includes}}
#import "{{this}}.h"
{{/includes}}

@interface {{config.classPrefix}}{{name}} ()
@end

@implementation {{config.classPrefix}}{{name}}

- (void)initProperty:(NSString *)prop {
{{#properties}}
  {{#if native}}
  {{else}}
  if ([prop isEqual:@"{{name}}"]) {
    self.{{name}} = {{type}}.new;
  }
  {{/if}}
{{/properties}}
}

- (void)initProperties {
}

@end

File: templates/android/Result.java
public class {{name}} {

    {{#properties}}
    {{#if comment}}
    /**
    {{{comment}}}
    */
    {{/if}}
    private {{type}} {{name}};
    {{/properties}}

    {{#properties}}
    public void set{{upperCaseName}}({{type}} {{name}}) {
        this.{{name}} = {{name}};
    }

    public {{type}} get{{upperCaseName}}() {
        return this.{{name}};
    }

    {{/properties}}
}
